### PR TITLE
Feat/process multi tx upgrades in cli

### PR DESCRIPTION
**Pull request by calvogenerico**
Improvements in CLI.

- Forking network with anvil to be able to execute multiple transactions for an upgrade.
- Fixed all commands to use the new upgrade file format. Including using multiple calls.
- Using debug calls to get changes in l2.

### CODE REVIEW

File: apps/cli/src/ethereum/contract-abi.ts [hunk 1]
```diff
@@ -2,13 +2,19 @@ import {
...
   type Hex,
   toEventSelector,
   toFunctionSelector,
 } from "viem";
 import type { z } from "zod";
+import { Option } from "nochoices";
+
+function isAbiFunction(abiElem: AbiItem): abiElem is AbiFunction {
+  return abiElem.type === "function";
+}
```
Comment (aon) [inline apps/cli/src/ethereum/contract-abi.ts:17]: Nice type guards use!

File: apps/cli/src/ethereum/contract-abi.ts [hunk 2]
```diff
@@ -65,4 +71,13 @@ export class ContractAbi {
   hasFunction(name: string): boolean {
     return this.raw.some((t) => t.type === "function" && t.name === name);
   }
+
+  selectorForFn(name: string): Option<Hex> {
+    const found = this.raw.find((t) => t.type === "function" && t.name === name);
+
+    return Option.fromNullable(found)
+      .filter(isAbiFunction)
+      .map((abiFn) => abiFn as AbiFunction)
```
Comment (aon) [inline apps/cli/src/ethereum/contract-abi.ts:80]: Is this necessary with the previous typeguard?

File: apps/cli/src/ethereum/rpc-client.ts [hunk 3]
```diff
@@ -208,38 +229,11 @@ export class RpcClient {
...
-    })) as any;
-
-    return {
-      number: block.number,
-      timestamp: hexToNumber(block.timestamp),
-    };
-  }
+  // async execDebugTx(protocolHandlerAddress: Address, target: Address, data: Hex, value: bigint): Promise<void> {
+  //
+  // }
```
Comment (aon) [inline apps/cli/src/ethereum/rpc-client.ts:238]:
```suggestion
```

**Review by aon — APPROVED** [review]

File: apps/cli/hardhat.config.cts [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import type { HardhatUserConfig } from "hardhat/config";
+
+if (process.env.RICH_PRIVATE_KEY === undefined) {
```
[the remark this answers is not in the document]
Comment (calvogenerico) [reply apps/cli/hardhat.config.cts:3]: The idea of the name was to highlight that it's an address with money inside. I changed it to "FUNDED_*"

File: apps/cli/src/ethereum/contract-abi.ts (hunk 2 above)
Comment (calvogenerico) [reply apps/cli/src/ethereum/contract-abi.ts]: I refactored this to make better use of the type predicate and try to make it cleaner.

**Pull request merged**